pkg/controllers/edgeconnect/controller.go

```diff
@@ -810,19 +817,31 @@ func (controller *Controller) createOrUpdateEdgeConnectConfigSecret(ctx context.
 	)
 
 	if err != nil {
+		conditions.SetSecretGenFailed(ec.Conditions(), consts.SecretConfigConditionType, err)
+
 		return "", "", errors.WithStack(err)
 	}
 
 	query := k8ssecret.Query(controller.client, controller.apiReader, log)
 
-	_, err = query.CreateOrUpdate(ctx, secretConfig)
+	created, err := query.CreateOrUpdate(ctx, secretConfig)
```

Comment:
the returned `bool` will be `true` on both the update and create scenario, it is `false` IF there was no change or an error happened

Reply:
fixed in 746c065f3